quickwit-janitor/src/janitor_service.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright (C) 2022 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use quickwit_actors::ActorHandle;
+use quickwit_metastore::MetastoreError;
+use quickwit_storage::StorageResolverError;
+use thiserror::Error;
+use tracing::error;
+
+use crate::actors::GarbageCollector;
+
+#[derive(Error, Debug)]
+pub enum JanitorServiceError {
```

Comment:
Having to redefine these error enum everywhere is a bit annoying, especially when there is the same type of errors. I have no magic solution right now but I put it here just to share my thoughts :)

Reply:
My rule of thumb is: don't create custom errors if you're not going to use them. I'd remove them for now.